### PR TITLE
prompt user to save project before closing

### LINKED ISSUE #149: Prompt user to save work if closing with unsaved changes
Notify the user if they are closing the app with unsaved or unpublished changes. Allow them to save or publish from the prompt.

### PR DESCRIPTION
**Pull request by mdaxtman**
prompts user to save if changes have been made to the project. Fixes #149  pending conversation regarding "publishing"

### CODE REVIEW

File: main.development.js [hunk 1]
```diff
@@ -181,6 +182,25 @@ app.on("ready", () => {
   });
 
   mainWindow.on("close", (ev) => {
+    if (promptToSave) {
+      ev.preventDefault();
```
Comment (bmathews) [inline main.development.js]: trying to quit the whole app with `cmd+q` doesn't happen if you press don't save

File: main.development.js [hunk 2]
```diff
@@ -233,6 +253,10 @@ app.on("ready", () => {
     });
   });
 
+  ipcMain.on("save-prompt", (event, saveRequired) => {
```
Comment (bmathews) [inline main.development.js]: Can you change the `save-prompt` message name to something like `dirty-state-changed` instead? was confused by the name at first

File: main.development.js [hunk 3]
```diff
@@ -181,6 +182,25 @@ app.on("ready", () => {
   });
 
   mainWindow.on("close", (ev) => {
+    if (promptToSave) {
+      ev.preventDefault();
+
+      dialog.showMessageBox({
+        type: "question",
+        buttons: ["Save", "Don't save", "Cancel"],
+        message: "Do You Wish to Save Your Project Before Quitting?"
```
Comment (bmathews) [inline main.development.js]: Can you change this to sentence case? `Do you wish to save your project before quitting?`

**Comment by bmathews** [comment]
🚀

**Comment by bmathews** [comment]
:shipit:

**Comment by mdaxtman** [comment]
For reference, this is the Windows dialog box:

capture

**Pull request merged**